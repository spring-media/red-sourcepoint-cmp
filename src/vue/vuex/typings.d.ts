import { CustomPurpose, CustomVendor } from '../../sourcepoint/typings';

export type SourcepointModuleState = {
  consentedCustomVendors: CustomVendor[];
  consentedCustomPurposes: CustomPurpose[];
  grantedVendors: string[];
  consentReady: boolean;
};
