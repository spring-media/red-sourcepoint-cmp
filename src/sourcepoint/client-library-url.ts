export const getClientLibraryUrl = (): string => `https://cdn.privacy-mgmt.com/wrapperMessagingWithoutDetection.js`;
