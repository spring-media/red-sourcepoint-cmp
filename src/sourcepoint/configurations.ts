import { ConfigurationParameters, SourcepointConfigurationObject } from './typings';

export const createConfiguration = (parameters: ConfigurationParameters): Partial<SourcepointConfigurationObject> => ({
  config: {
    accountId: parameters.accountId,
    wrapperAPIOrigin: 'https://wrapper-api.sp-prod.net/tcfv2',
    propertyId: parameters.propertyId,
    mmsDomain: 'https://message.sp-prod.net',
    events: parameters.events,
  },
});
