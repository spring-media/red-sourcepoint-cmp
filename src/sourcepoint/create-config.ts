import { Config } from './typings';

export const createConfig = (parameters: Config): Config => ({
  accountId: 0,
  wrapperAPIOrigin: 'https://wrapper-api.sp-prod.net/tcfv2',
  propertyId: 0,
  mmsDomain: 'https://message.sp-prod.net',
  ...parameters,
});
