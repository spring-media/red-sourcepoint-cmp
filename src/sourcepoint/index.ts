export * from './iab-stub-script';
export * from './client-library-url';
export * from './has-custom-consent';
export * from './privacy-manager';
export * from './create-config';
export * from './get-custom-vendor-consents';
