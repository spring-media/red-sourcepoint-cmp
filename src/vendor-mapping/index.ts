export * from './vendors';
export * from './purposes';
export * from './api';
