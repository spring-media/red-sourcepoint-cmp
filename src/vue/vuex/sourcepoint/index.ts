import { Commit, Dispatch, Module } from 'vuex';
import { getCustomVendorConsents } from '../../../sourcepoint';
import { SourcepointModuleState } from '../typings';
import { CustomPurpose, CustomVendor } from '../../../sourcepoint/typings';
import {
  configureGrants,
  configureVendorPurposeMapping,
  getGrantedVendors,
  loadVendorPurposeMapping,
} from '../../../vendor-mapping';

const moduleState: SourcepointModuleState = {
  consentedCustomVendors: [],
  consentedCustomPurposes: [],
  grantedVendors: [],
};

export const mutations = {
  setCustomVendorConsents(state: SourcepointModuleState, payload: CustomVendor[]): void {
    state.consentedCustomVendors = payload;
  },
  setCustomPurposeConsents(state: SourcepointModuleState, payload: CustomPurpose[]): void {
    state.consentedCustomPurposes = payload;
  },
  setGrantedVendors(state: SourcepointModuleState, payload: string[]): void {
    state.grantedVendors = payload;
  },
};

export const getters = {
  vendorHasConsent: (state: SourcepointModuleState) => (vendorId: string): boolean =>
    state.grantedVendors.includes(vendorId),
};

export const actions = {
  async refreshCustomVendorConsents({ commit }: { commit: Commit }): Promise<void> {
    const { consentedPurposes = [], consentedVendors = [], grants = {} } = await getCustomVendorConsents();

    configureGrants(grants);

    commit('setCustomVendorConsents', consentedVendors);
    commit('setCustomPurposeConsents', consentedPurposes);
    commit('setGrantedVendors', getGrantedVendors());
  },
  async bootstrapConsents({ dispatch }: { dispatch: Dispatch }, payload: { propertyId: number }): Promise<void> {
    await dispatch('refreshCustomVendorConsents');

    const mappings = await loadVendorPurposeMapping(payload.propertyId);

    configureVendorPurposeMapping(mappings);
  },
};

export const sourcepoint: Module<SourcepointModuleState, Record<string, unknown>> = {
  namespaced: true,
  state: moduleState,
  mutations,
  actions,
  getters,
};
