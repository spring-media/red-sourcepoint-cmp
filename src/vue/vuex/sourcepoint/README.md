# sourcepoint Vuex Module

## Effects

### reloadPageOnReject

This effect reloads the page after a vendor or purpose has been rejected.

## Example

```javascript
import Vue from 'vue';
import Vuex from 'vuex';
import { sourcepoint } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/vuex/sourcepoint';
import { reloadPageOnReject } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/vuex/sourcepoint/effects';

Vue.use(Vuex);

const store = new Vuex.Store({
  strict: true,
  state: {},
  modules: {
    sourcepoint,
  },
});

reloadPageOnReject(store);
``` 
