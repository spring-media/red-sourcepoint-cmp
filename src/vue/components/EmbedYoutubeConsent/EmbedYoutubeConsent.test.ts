import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import { EmbedYoutubeConsent } from './';
import { sourcepoint } from '../../vuex/sourcepoint';
import { PURPOSE_ID_SOCIAL, VENDOR_ID_YOUTUBE } from '../../../vendor-mapping';

const localVue = createLocalVue();

localVue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    sourcepoint,
  },
});

describe('EmbedYoutubeConsent component', () => {
  afterEach(() => {
    store.commit('sourcepoint/setCustomVendorConsents', []);
    store.commit('sourcepoint/setCustomPurposeConsents', []);
  });

  it('should render the placeholder by default', () => {
    const wrapper = mount(EmbedYoutubeConsent, {
      propsData: {
        privacyManagerId: 12345,
      },
      store,
      localVue,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render the embed in case of a consented vendor and his purpose(s)', () => {
    store.commit('sourcepoint/setCustomVendorConsents', [{ _id: VENDOR_ID_YOUTUBE }]);
    store.commit('sourcepoint/setCustomPurposeConsents', [{ _id: PURPOSE_ID_SOCIAL }]);

    const wrapper = mount(EmbedYoutubeConsent, {
      propsData: {
        privacyManagerId: 12345,
        content: '<div>Youtube Embed Content</div>',
      },
      store,
      localVue,
    });

    expect(wrapper.element).toMatchInlineSnapshot(`
      <div>
        <div>
          Youtube Embed Content
        </div>
      </div>
    `);
  });

  it('should load the privacy manager', () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    window._sp_ = {
      loadPrivacyManagerModal: jest.fn(),
    };

    const wrapper = mount(EmbedYoutubeConsent, {
      propsData: {
        privacyManagerId: 12345,
        content: '<div>Youtube Embed Content</div>',
      },
      store,
      localVue,
    });

    wrapper.find('.embed-placeholder__button').trigger('click');

    expect(window._sp_.loadPrivacyManagerModal).toHaveBeenCalledWith(12345);
  });
});
