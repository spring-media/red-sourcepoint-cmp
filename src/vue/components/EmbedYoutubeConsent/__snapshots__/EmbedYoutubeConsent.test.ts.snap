// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedYoutubeConsent component should render the placeholder by default 1`] = `
<div
  class="embed-placeholder__container"
  data-vendor-id="5e7ac3fae30e7d1bc1ebf5e8"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="172"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        style="transform: translate(calc(50% - 20px), calc(50% - 20px)) scale(0.42);"
      >
        <path
          d="M 118.9,13.3 C 117.5,8.1 113.4,4 108.2,2.6 98.7,0 60.7,0 60.7,0 60.7,0 22.7,0 13.2,2.5 8.1,3.9 3.9,8.1 2.5,13.3 0,22.8 0,42.5 0,42.5 0,42.5 0,62.3 2.5,71.7 3.9,76.9 8,81 13.2,82.4 22.8,85 60.7,85 60.7,85 c 0,0 38,0 47.5,-2.5 5.2,-1.4 9.3,-5.5 10.7,-10.7 2.5,-9.5 2.5,-29.2 2.5,-29.2 0,0 0.1,-19.8 -2.5,-29.3 z"
          fill="#E9ECEF"
        />
         
        <polygon
          fill="#ffffff"
          points="80.2,42.5 48.6,24.3 48.6,60.7"
        />
      </g>
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
    
    An dieser Stelle findest du Inhalte aus YouTube
  
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
    Um mit Inhalten aus YouTube und anderen sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen wir
    deine Zustimmung.
  
  </div>
   
  <button
    class="embed-placeholder__button"
  >
    
    soziale Netzwerke aktivieren
  
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir Inhalte von Drittanbietern angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Dazu ist ggf. die Speicherung von Cookies auf
        deinem Gerät notwendig. Mehr Informationen dazu findest du
        
    <a
      class="embed-placeholder__text-link"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      hier
    </a>
    .
      
  </div>
</div>
`;
